src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs

```diff
@@ -0,0 +1,464 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.Recommender;
+using Microsoft.ML.Runtime.Recommender.Internal;
+using Microsoft.ML.Runtime.Model.Pfa;
+using Microsoft.ML.Runtime.Model.Onnx;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data.IO;
+
+[assembly: LoadableClass(typeof(MatrixFactorizationPredictor), null, typeof(SignatureLoadModel), "Matrix Factorization Predictor Executor", MatrixFactorizationPredictor.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(MatrixFactorizationPredictionTransformer), typeof(MatrixFactorizationPredictionTransformer),
+    null, typeof(SignatureLoadModel), "", MatrixFactorizationPredictionTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Recommender
+{
+    public sealed class MatrixFactorizationPredictor : IPredictor, ICanSaveModel, ICanSaveInTextFormat, ISchemaBindableMapper
+    {
+        public const string LoaderSignature = "MFPredictor";
+        public const string RegistrationName = "MatrixFactorizationPredictor";
```

Comment:
can those be internal?

Reply:
Sure.

---
In reply to: [226814020](https://github.com/dotnet/machinelearning/pull/1263#discussion_r226814020) [](ancestors = 226814020)